controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -120,8 +124,24 @@
     public CompletableFuture<UpdateStreamStatus> sealStream(final String scope, final String stream) {
         Exceptions.checkNotNullOrEmpty(scope, "scope");
         Exceptions.checkNotNullOrEmpty(stream, "stream");
-        return streamMetadataTasks.sealStream(scope, stream, null)
-                .thenApplyAsync(status -> UpdateStreamStatus.newBuilder().setStatus(status).build(), executor);
+        return streamStore.getActiveSegments(scope, stream, null, executor).thenCompose(segments -> {
+            if (segments.isEmpty()) {
+                return CompletableFuture.completedFuture(ScaleResponse.newBuilder()
+                        .setStatus(ScaleResponse.ScaleStreamStatus.SUCCESS).build());
+            } else {
+                return streamMetadataTasks.manualScale(scope,
+                        stream,
+                        segments.stream().map(Segment::getNumber).collect(Collectors.toList()),
+                        Collections.emptyList(),
```

Comment:
there might be an input validation check in scale operation that verifies Union(segmentsToSeal.keyRanges) == Union(newSegments.keyRanges).
here if we are passing new Ranges as Collections.emptyList(), we will need to special case it.